### PR TITLE
Improve configuration variables name and documentation

### PR DESCRIPTION
**Pull request by avvertix**
This pull request want to uniform the naming schema across all environment variables that govern the K-Box customization.

The new naming schema uses the `KBOX_` for all variables.

In addition, this pull request, add linting of shell scripts via [shellcheck](https://www.shellcheck.net/)

Here is a summary of the changes (extracted from the configuration.md file)

| version 0.21 and below                | new name from version 0.22         |
|---------------------------------------|------------------------------------|
| `KLINK_DMS_APP_URL`  | `KBOX_APP_URL`                     |
| `KLINK_DMS_APP_INTERNAL_URL`| `KBOX_APP_LOCAL_URL`               |
| `KLINK_DMS_APP_KEY`                   | `KBOX_APP_KEY`                     |
| `KLINK_DMS_APP_ENV`                   | `KBOX_APP_ENV`                     |
| `KLINK_DMS_APP_DEBUG`                 | `KBOX_APP_DEBUG`                   |
| `KLINK_DMS_ADMIN_USERNAME`            | `KBOX_ADMIN_USERNAME`              |
| `KLINK_DMS_ADMIN_PASSWORD`            | `KBOX_ADMIN_PASSWORD`              |
| `KLINK_DMS_CORE_ADDRESS`              | `KBOX_SEARCH_SERVICE_URL`          |
| `KLINK_DMS_DB_NAME`                   | `KBOX_DB_NAME`                     |
| `KLINK_DMS_DB_HOST`                   | `KBOX_DB_HOST`                     |
| `KLINK_DMS_DB_USERNAME`               | `KBOX_DB_USERNAME`                 |
| `KLINK_DMS_DB_PASSWORD`               | `KBOX_DB_PASSWORD`                 |
| `KLINK_DMS_DB_TABLE_PREFIX`           | `KBOX_DB_TABLE_PREFIX`             |
| `KLINK_DMS_MAX_UPLOAD_SIZE`           | `KBOX_UPLOAD_LIMIT`                |
| `KLINK_PHP_POST_MAX_SIZE`             | `KBOX_PHP_POST_MAX_SIZE`           |
| `KLINK_PHP_UPLOAD_MAX_FILESIZE`       | `KBOX_PHP_UPLOAD_MAX_FILESIZE`     |
| `KLINK_PHP_MEMORY_LIMIT`              | `KBOX_PHP_MEMORY_LIMIT`            |
| `DMS_ITEMS_PER_PAGE`                  | `KBOX_PAGE_LIMIT`                  |
| `MAIL_DRIVER`                         | `KBOX_MAIL_DRIVER`                 |
| `MAIL_HOST`                           | `KBOX_MAIL_HOST`                   |
| `MAIL_PORT`                           | `KBOX_MAIL_PORT`                   |
| `MAIL_FROM_ADDRESS`                   | `KBOX_MAIL_FROM_ADDRESS`           |
| `MAIL_FROM_NAME`                      | `KBOX_MAIL_FROM_NAME`              |
| `MAIL_ENCRYPTION`                     | `KBOX_MAIL_ENCRYPTION`             |
| `MAIL_USERNAME`                       | `KBOX_MAIL_USERNAME`               |
| `MAIL_PASSWORD`                       | `KBOX_MAIL_PASSWORD`               |
| `DMS_ARE_GUEST_PUBLIC_SEARCH_ENABLED` | `KBOX_ENABLE_GUEST_NETWORK_SEARCH` |

### Deprecation

The following environment variables are deprecated:

- `KLINK_DMS_DIR`
- `DMS_USE_HTTPS` is now deprecated and should not be set manually
- `DMS_INSTITUTION_IDENTIFIER` is now deprecated, and as part of the Institution management removal will be removed in a future version
- `DMS_IDENTIFIER` is now deprecated. The variable will be ignored if set
- `KLINK_SETUP_WWWUSER` is now deprecated, the default user will be `www-data`
- `DMS_ENABLE_ACTIVITY_TRACKING` was not used, so it has been deprecated. The variable will be ignored if set
- `DMS_UPLOAD_FOLDER` was not used, so it has been deprecated
- `DMS_RECENT_TIMELIMIT` is now deprecated as it was not used

### Changes

- `KBOX_APP_LOCAL_URL` default value, if used from Docker, is now `http://kbox/`. The new default value follow the example docker compose file
- `KBOX_PHP_POST_MAX_SIZE` and `KBOX_PHP_UPLOAD_MAX_FILESIZE` are automatically set based on the `KBOX_UPLOAD_LIMIT` value

### CODE REVIEW

**Comment by ailox** [comment]
LGTM, was able to test and verify that this works. I like the new naming scheme!

File: config/dms.php [hunk 1]
```diff
@@ -52,7 +54,7 @@
     |
     | @var boolean
     */
-    'are_guest_public_search_enabled' => getenv('DMS_ARE_GUEST_PUBLIC_SEARCH_ENABLED') ?: true,
+    'are_guest_public_search_enabled' => env('KBOX_ENABLE_GUEST_NETWORK_SEARCH', env('DMS_ARE_GUEST_PUBLIC_SEARCH_ENABLED', true)),
```
Comment (ailox) [inline config/dms.php]: Will we remember to un-wrap the redundant variables in a couple of versions from now? What's your view on how long to support this?

Comment (avvertix) [reply config/dms.php]: I guess we will remember, my idea is to have two versions before removing the support, so we will be sure we don't need to test anything below 0.22

File: docker/start.sh [hunk 2]
```diff
@@ -1,23 +1,10 @@
 #!/bin/bash
 
-DIR=$(pwd)
 COMMAND=${1}
 
 echo "K-Box start in progress..."
 
 case "${COMMAND}" in
```
Comment (ailox) [inline docker/start.sh]: Probably at this point we can drop the switch statement on ${COMMAND}?

Comment (avvertix) [reply docker/start.sh]: probably yes, there is no reason to do have it

**Comment by avvertix** [comment]
thanks @ailox I will remove the redundant `switch`

**Comment by avvertix** [comment]
@ailox I changed the start.sh script

**Pull request merged**